#VRML_SIM R2020b utf8
WorldInfo {
  basicTimeStep 8
}
Viewpoint {
  orientation -0.13541834677885825 0.8806310468065589 0.45403835824307326 2.815646252922907
  position 0.3629607025297309 4.815316860348194 -3.6396516306266213
  follow "ChuhaBasic"
  followType "None"
}
TexturedBackground {
  texture "mars"
}
TexturedBackgroundLight {
  texture "mars"
  castShadows FALSE
}
RectangleArena {
  translation -0.0320768 0 0.0713054
  floorSize 10 10
  floorAppearance Parquetry {
    type "light strip"
  }
}
WoodenBox {
  translation -0.66 0.3 0.68
}
WoodenBox {
  translation 0.66 0.3 0.68
  name "wooden box(1)"
}
Robot {
  translation 0.14652426598274915 0.0074579536914906475 0.02214180003386924
  rotation -0.000505407601181037 0.9999993469784033 -0.0010249906944676754 2.94682937449885
  children [
    InertialUnit {
      translation 0 0.0075 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0.5 0
            roughness 1
          }
          geometry Box {
            size 0.01 0.005 0.01
          }
        }
      ]
      boundingObject Shape {
        appearance PBRAppearance {
          baseColor 0 0.5 0
          roughness 1
        }
        geometry Box {
          size 0.01 0.005 0.01
        }
      }
    }
    Display {
      name "extra_display"
      width 1024
      height 1024
    }
    Lidar {
      translation 0 0.02 0
      rotation 0 0.9999999999999999 0 3.14
      children [
        DEF LIDAR_SHAPE Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
        }
      ]
      boundingObject USE LIDAR_SHAPE
      physics Physics {
      }
      rotationStep 0.262
      tiltAngle -0.12
      fieldOfView 0.3
      verticalFieldOfView 0.3
      numberOfLayers 16
      maxRange 6
      type "rotating"
    }
    Camera {
      translation 0 0 0.03
      rotation 0 1 0 -3.14
      children [
        USE LIDAR_SHAPE
      ]
      name "front_camera"
      boundingObject USE LIDAR_SHAPE
      physics Physics {
      }
      fieldOfView 2
      width 256
      height 256
      far 10
    }
    Camera {
      translation 0 0 -0.03
      children [
        USE LIDAR_SHAPE
      ]
      name "rear_camera"
      boundingObject USE LIDAR_SHAPE
      physics Physics {
      }
      fieldOfView 2
      width 256
      height 256
      far 10
    }
    DEF CHASSIS_GROUP Group {
      children [
        Transform {
          translation 0 0.0125 0
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.5 0.5 0.5
                roughness 0.5
              }
              geometry Cylinder {
                height 0.01
                radius 0.0025
              }
            }
          ]
        }
        DEF CHASSIS_SHAPE Shape {
          appearance PBRAppearance {
            baseColor 0.7 0 0
            metalness 0.7
          }
          geometry Cylinder {
            height 0.0148
            radius 0.03
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 15509.328167523308
        anchor 0.035 0 0
      }
      device [
        PositionSensor {
          name "left motor sensor"
        }
        RotationalMotor {
          name "left motor"
          maxVelocity 60
          maxTorque 0.05
        }
      ]
      endPoint Solid {
        translation 0.034999999740545906 4.441317810577016e-05 3.3220518961824615e-05
        rotation -0.5145628085168344 0.5145646983648784 0.685892329223712 1.9392063890708489
        children [
          DEF WHEEL_SHAPE Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.005
              radius 0.0075
            }
          }
        ]
        name "LeftWheelSolid"
        boundingObject USE WHEEL_SHAPE
        physics DEF WHEEL_PHY Physics {
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 12142.621981233602
        anchor -0.055 0 0
      }
      device [
        PositionSensor {
          name "right motor sensor"
        }
        RotationalMotor {
          name "right motor"
          maxVelocity 60
          maxTorque 0.05
        }
      ]
      endPoint Solid {
        translation -0.03500000038240844 -4.006410831159118e-05 -3.79618322983081e-05
        rotation -0.37995902000756615 0.3799600802457111 -0.8433631960991437 4.542846992722221
        children [
          USE WHEEL_SHAPE
        ]
        name "RightWheelSolid"
        boundingObject USE WHEEL_SHAPE
        physics USE WHEEL_PHY
      }
    }
  ]
  name "ChuhaCameraLidar"
  boundingObject USE CHASSIS_GROUP
  physics Physics {
    mass 0.3
  }
  controller "lidar_camera_controller"
  linearVelocity 6.232266328806033e-13 -2.0166048513873838e-14 -1.2471853620901518e-13
  angularVelocity 6.776746323111518e-13 9.773333527674542e-11 2.2766058360531475e-14
}
Robot {
  translation 1.4506117227896504 0.007485974495702329 -0.24726372008378755
  rotation 0.0034541495527834797 -0.9999938839939287 -0.0005484757093350092 -0.314945997882939
  children [
    InertialUnit {
      translation 0 0.0075 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0.5 0
            roughness 1
          }
          geometry Box {
            size 0.01 0.005 0.01
          }
        }
      ]
      boundingObject Shape {
        appearance PBRAppearance {
          baseColor 0 0.5 0
          roughness 1
        }
        geometry Box {
          size 0.01 0.005 0.01
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 14860.169055217251
        anchor 0.035 0 0
      }
      device [
        PositionSensor {
          name "left motor sensor"
        }
        RotationalMotor {
          name "left motor"
          maxVelocity 60
          maxTorque 0.05
        }
      ]
      endPoint Solid {
        translation 0.035 0 0
        rotation -0.6972050585643901 0.6972076195462713 0.1667532355273725 2.81112803469522
        children [
          DEF WHEEL_SHAPE Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.005
              radius 0.0075
            }
          }
        ]
        name "LeftWheelSolid"
        boundingObject USE WHEEL_SHAPE
        physics DEF WHEEL_PHY Physics {
          mass 0.05
        }
        linearVelocity 5.155456630622013e-12 6.899200557484994e-18 1.5818179443671795e-11
        angularVelocity 2.1091994277001056e-09 8.022824377930778e-15 -6.871531633454325e-10
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 11575.672580723556
        anchor -0.055 0 0
      }
      device [
        PositionSensor {
          name "right motor sensor"
        }
        RotationalMotor {
          name "right motor"
          maxVelocity 60
          maxTorque 0.05
        }
      ]
      endPoint Solid {
        translation -0.03500000000000002 0 0
        rotation 0.25981969804822713 -0.25982065242102237 0.9300467477937022 1.6432569347427253
        children [
          USE WHEEL_SHAPE
        ]
        name "RightWheelSolid"
        boundingObject USE WHEEL_SHAPE
        physics USE WHEEL_PHY
        linearVelocity 5.1518806936603525e-12 -1.123946464490311e-18 1.581937131720087e-11
        angularVelocity 2.1091991218957105e-09 9.70023270432667e-15 -6.871544383727389e-10
      }
    }
    DEF CHASSIS_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.7 0 0
        metalness 0.7
      }
      geometry Cylinder {
        height 0.0149
        radius 0.03
      }
    }
  ]
  name "ChuhaBasic"
  boundingObject USE CHASSIS_SHAPE
  physics Physics {
  }
  controller "teleoperator"
  linearVelocity 5.153550113196886e-12 -2.1082473652919163e-17 1.5819452311676227e-11
  angularVelocity 2.1092009942397603e-09 3.8593730052683804e-15 -6.871531677295935e-10
}
